engines/asylum/resources/encounters.cpp

```diff
@@ -0,0 +1,1688 @@
+			done = true;
+			_value1 = 0;
+			break;
+
+		case kOpcodeSetScriptVariable:
+			_scriptData.vars[entry.param1] = getVariableInv(entry.param2);
+			break;
+
+		case kOpcodeSetCounterFromVars:
+			_scriptData.counter = _scriptData.vars[entry.param1] - getVariableInv(entry.param2);
+			break;
+
+		case kOpcodeSetOffset:
+			_scriptData.offset = entry.param2 - 1;
+			break;
+
+		case kOpcodeSetOffsetIfCounterNegative:
+			if (_scriptData.counter >= 0)
+				break;
+
+			_scriptData.offset = entry.param2 - 1;
+			break;
+
+		case kOpcodeSetOffsetIfCounterNegativeOrNull:
+			if (_scriptData.counter > 0)
+				break;
+
+			_scriptData.offset = entry.param2 - 1;
+			break;
+
+		case kOpcodeSetOffsetIfCounterIsNull:
+			if (_scriptData.counter)
+				break;
+
+			_scriptData.offset = entry.param2 - 1;
+			break;
+
+		case kOpcodeSetOffsetIfCounterIsNotNull:
+			if (!_scriptData.counter)
+				break;
+
+			_scriptData.offset = entry.param2 - 1;
+			break;
+
+		case kOpcodeSetOffsetIfCounterPositiveOrNull:
+			if (_scriptData.counter < 0)
+				break;
+
+			_scriptData.offset = entry.param2 - 1;
+			break;
+
+		case kOpcodeSetOffsetIfCounterPositive:
+			if (_scriptData.counter <= 0)
+				break;
+
+			_scriptData.offset = entry.param2 - 1;
+			break;
+
+		case kOpcodeSetCurrentItemOptions:
+			if (entry.param1)
+				_item->keywords[findKeyword(_item, entry.param2)] &= -((kKeywordOptionsDisabled << 8) + 1);
+			else
+				_item->keywords[findKeyword(_item, entry.param2)] |= (kKeywordOptionsVisible << 8);
+			break;
+
+		case kOpcodeClearCurrentItemOptions:
+			if (entry.param1)
+				_item->keywords[findKeyword(_item, entry.param2)] |= (kKeywordOptionsDisabled << 8);
+			else
+				_item->keywords[findKeyword(_item, entry.param2)] &= -((kKeywordOptionsVisible << 8) + 1);
+			break;
+
+		case kOpcodeSetItemOptions:
+			_items[entry.param1].keywords[findKeyword(&_items[entry.param1], entry.param2)] |= (kKeywordOptionsUnknown << 8);
+			_items[entry.param1].keywords[findKeyword(&_items[entry.param1], entry.param2)] |= (kKeywordOptionsVisible << 8);
+
+			// Original saves the item back here
+			break;
+
+		case kOpcodeCloseDialog:
+			if (!_isClosing)
+				_shouldCloseDialog = true;
+
+			done = true;
+			break;
+
+		case kOpcodeResetSpeech:
+			resetSpeech(_item->speechResourceId, getVariableInv(entry.param2));
+
+			done = true;
+			break;
+
+		case kOpcodeSetVariable:
+			setVariable(entry.param2, (int16)_scriptData.vars[entry.param1]);
+			break;
+
+		case kOpcodeIncrementScriptVariable:
+			_scriptData.vars[entry.param1] += entry.param2;
+			break;
+
+		case kOpcodeProcessVariable3:
+			switch (getVariable(3)) {
+			default:
+				break;
+
+			case 13:
+			case 15:
+				if (_actorIndex) {
+					getScene()->getActor(_actorIndex)->hide();
+				} else {
+					getWorld()->getObjectById(_objectId1)->disable();
+					getWorld()->getObjectById(_objectId2)->disable();
+				}
+
+				if (getVariable(3) == 13)
+					break;
+				// Fallback to next case
```

Comment:
Compiler Warning:
````
engines/asylum/resources/encounters.cpp: In member function ‘void Asylum::Encounter::runScript()’:
engines/asylum/resources/encounters.cpp:1605:5: warning: this statement may fall through [-Wimplicit-fallthrough=]
     if (getVariable(3) == 13)
     ^~
engines/asylum/resources/encounters.cpp:1609:4: note: here
    case 14:
    ^~~~
````
Again, this looks intentional, so change or add // fallthrough to remove the warning.